engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -649,6 +650,32 @@ protected void serviceStart() throws Exception
     }
   }
 
+  /**
+   * Modify the config object by adding SSL related parameters in a resource as the case may be so the WebApp uses
+   * correct SSL params
+   *
+   * @param config  Configuration to be modified
+   */
+  private void addSSLConfigResource(Configuration config)
+  {
+    SSLConfig sslConfig = dag.getValue(Context.DAGContext.SSL_CONFIG);
+    if (sslConfig != null) {
+      String nodeLocalConfig = sslConfig.getNodeLocalConfigPath();
+      if (StringUtils.isNotEmpty(nodeLocalConfig)) {
+        config.addResource(new Path(nodeLocalConfig));
+      } else {
+        // create a configuration object and add it as a resource
+        Configuration sslConfigResource = new Configuration(false);
+        String keyStorePath = sslConfig.getKeyStorePath();  // need to strip out the root
+        keyStorePath = (new File(keyStorePath)).getName();
+        sslConfigResource.set("ssl.server.keystore.location", new File(keyStorePath).getName(), Context.DAGContext.SSL_CONFIG.getLongName());
```

Comment:
Define ssl.server.keystore.location as a constant and add a note to revisit after Hadoop dependency upgrade.